### PR TITLE
Support custom LLM implementation

### PR DESCRIPTION
**Pull request by nherment**
This adds the ability to inject your own LLM into holmesgps when you invoke it as a library.

### CODE REVIEW

File: holmes/core/tool_calling_llm.py [hunk 1]
```diff
@@ -147,26 +89,21 @@ def call(self, system_prompt, user_prompt, post_process_prompt: Optional[str] =
...
             logging.debug(f"sending messages {messages}")
             try:
-                full_response = litellm.completion(
-                    model=self.model,
-                    api_key=self.api_key,
+                full_response = self.llm.completion(
                     messages=messages,
                     tools=tools,
                     tool_choice=tool_choice,
-                    base_url=self.base_url,
```
Comment (arikalon1) [inline holmes/core/tool_calling_llm.py:184]: why did we remove the base_url, temperature and drop_params?

**Review by arikalon1 — COMMENTED** [review]
nice work

I think we missed something.
This [method](https://github.com/robusta-dev/holmesgpt/blob/830bcec379ad1b659f85879ef1d972fdcae5cd41/holmes/config.py#L143), still returns `ToolCallingLLM`

I think it causes an issue [here](https://github.com/robusta-dev/holmesgpt/blob/830bcec379ad1b659f85879ef1d972fdcae5cd41/server.py#L203) because it doesn't contain the required methods anymore

WDYT?

File: holmes/core/tool_calling_llm.py (hunk 1 above)
Comment (nherment) [reply holmes/core/tool_calling_llm.py:184]: Fixed temp & drop_params. base_url is implementated in the `DefaultLLM`

**Comment by nherment** [comment]
>I think we missed something.
> This [method](https://github.com/robusta-dev/holmesgpt/blob/830bcec379ad1b659f85879ef1d972fdcae5cd41/holmes/config.py#L143), still returns ToolCallingLLM

> I think it causes an issue [here](https://github.com/robusta-dev/holmesgpt/blob/830bcec379ad1b659f85879ef1d972fdcae5cd41/server.py#L203) because it doesn't contain the required methods anymore

The code has been updated to reach into the llm object of the toolCalling instance:
image

**Review by arikalon1 — APPROVED** [review]
nice work

**Pull request merged**